src/runtime/Codecs/FunctionCodec.cs

```diff
@@ -0,0 +1,377 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace Python.Runtime.Codecs
+{
+    //like MethodWrapper but not static, so we can throw some state into it.
+    internal class MethodWrapper2
+    {
+        public IntPtr mdef;
+        public IntPtr ptr;
+        private bool _disposed = false;
+        private ThunkInfo _thunk;
+
+        public MethodWrapper2(object instance, string name, string funcType = null)
+        {
+            // Turn the managed method into a function pointer
+            var type = instance.GetType();
+            _thunk = GetThunk(instance, type.GetMethod(name), funcType);
+
+            // Allocate and initialize a PyMethodDef structure to represent
+            // the managed method, then create a PyCFunction.
+
+            mdef = Runtime.PyMem_Malloc(4 * IntPtr.Size);
+            TypeManager.WriteMethodDef(mdef, name, _thunk.Address, 0x0003);
+            ptr = Runtime.PyCFunction_NewEx(mdef, IntPtr.Zero, IntPtr.Zero);
+        }
```

Comment:
I though we already have a delegate -> Python callable conversion. What is different about this one?
Same goes for method groups: you can just do `dotNetInstance.ToPython().GetAttr('NameOfGroupOfMethods')`.

Reply:
For example: 42.ToPython().GetAttr('ToString')

Reply:
@lostmsu I didn't know about it, I looked for PyCFunction_* and this is how I ended up finding the original MethodWrapper.  I'll see if I can find out how to do what you describe

Reply:
@lostmsu ToPython() uses Python.Runtime.Converter which uses codecs but falls back to GetInstHandle for TypeCode.Object (delegate types use that TypeCode). I guess if this already works I do not need to implement IPyObjectEncoder in my codec.  Let me try and write an embedding test to see what the problem is if there is one.